File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/storage/TripDetailsStorage.java
package com.worldventures.dreamtrips.modules.trips.storage;

import android.support.annotation.Nullable;

import com.worldventures.dreamtrips.core.janet.cache.CacheBundle;
import com.worldventures.dreamtrips.core.janet.cache.CachedAction;
import com.worldventures.dreamtrips.core.janet.cache.storage.ActionStorage;
import com.worldventures.dreamtrips.core.repository.SnappyRepository;
import com.worldventures.dreamtrips.modules.trips.command.GetTripDetailsCommand;
import com.worldventures.dreamtrips.modules.trips.model.TripModel;

public class TripDetailsStorage implements ActionStorage<TripModel> {

   public static final String UID = "UID";

   private final SnappyRepository snappyRepository;

   public TripDetailsStorage(SnappyRepository snappyRepository) {
      this.snappyRepository = snappyRepository;
   }

   @Override
   public Class<? extends CachedAction> getActionClass() {
      return GetTripDetailsCommand.class;
   }

   @Override
   public void save(@Nullable CacheBundle params, TripModel data) {
      snappyRepository.saveTripDetails(data);
   }

   @Override
   public TripModel get(@Nullable CacheBundle action) {
      return snappyRepository.getTripDetail(action.get(UID));
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/service/storage/TripDetailsStorage.kt
package com.worldventures.dreamtrips.modules.trips.service.storage

import com.worldventures.dreamtrips.core.repository.SnappyRepository
import com.worldventures.dreamtrips.modules.trips.model.TripModel
import com.worldventures.dreamtrips.modules.trips.service.command.GetTripDetailsCommand
import com.worldventures.janet.cache.CacheBundle
import com.worldventures.janet.cache.storage.ActionStorage

class TripDetailsStorage(private val db: SnappyRepository) : ActionStorage<TripModel> {

   override fun getActionClass() = GetTripDetailsCommand::class.java

   override fun save(params: CacheBundle?, data: TripModel) = db.saveTripDetails(data)

   override fun get(action: CacheBundle?) = db.getTripDetail(action?.get(UID))

   companion object {
      val UID = "UID"
   }
}
